resources/assets/components/CampaignSingle/index.js

```diff
@@ -20,21 +22,30 @@ class CampaignSingle extends React.Component {
 
     this.api = new RestApiClient;
     this.filterPosts = this.filterPosts.bind(this);
-    this.getPostsByTag = this.getPostsByTag.bind(this);
-    this.getPostsByStatus = this.getPostsByStatus.bind(this);
+    this.getPostsByFilter = this.getPostsByFilter.bind(this);
     this.getPostsByPaginatedLink = this.getPostsByPaginatedLink.bind(this);
   }
 
   // Filter posts based on status or tag(s).
-  filterPosts(filter) {
-    // If the filter is a status, make API call to get posts by status.
-    if (['pending', 'accepted', 'rejected'].includes(filter)) {
-      this.getPostsByStatus(filter, this.props.campaign.id);
-    } else {
-      // If the filter is a tag, make the API call to get posts by tag.
-      this.getPostsByTag(filter, this.props.campaign.id);
+  filterPosts(state) {
+    // Grab all of the active tags to send to API request.
+    let activeTags = [];
+
+    if (state.tags) {
+      Object.keys(state.tags).forEach(function(key) {
+        if (state.tags[key] === true) {
+         activeTags.push(key);
+        }
+      });
     }
 
+    let filters = {
```

Comment:
Following my above comment, if `filters` is it's own object in `state` you would just be able to pass it to `getPostsByFilter()` or do whatever parsing work you need to do with the object to get it in the right format for the call.